pkg/steps/rpm_injection_test.go

```diff
@@ -0,0 +1,87 @@
+package steps
+
+import (
+	"testing"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakerest "k8s.io/client-go/rest/fake"
+
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+func TestRPMImageInjectionStep(t *testing.T) {
+	config := api.RPMImageInjectionStepConfiguration{
+		From: "configFrom",
+		To:   "configTo",
+	}
+	resources := api.ResourceConfiguration{}
+
+	fakecs := createTestingClient(t)
+
+	istclient := fakecs.imagecs.ImageV1()
+	routeclient := fakecs.routecs.RouteV1()
+	buildsgetter := fakecs.buildcs.BuildV1()
+	restclient := &fakerest.RESTClient{}
+	buildclient := NewBuildClient(buildsgetter, restclient)
+
+	jobspec := &api.JobSpec{Namespace: "jobspecNamespace"}
+
+	rpmiis := RPMImageInjectionStep(config, resources, buildclient, routeclient, istclient, jobspec)
+
+	specification := stepExpectation{
+		name: "configTo",
+		requires: []api.StepLink{
+			api.InternalImageLink("configFrom"),
+			api.RPMRepoLink(),
+		},
+		creates:  []api.StepLink{api.InternalImageLink("configTo")},
+		provides: providesExpectation{params: nil, link: nil},
+		inputs:   inputsExpectation{values: nil, err: false},
+	}
+
+	examineStep(t, rpmiis, specification)
+
+	rpmroute := &routev1.Route{
+		ObjectMeta: meta.ObjectMeta{Name: RPMRepoName},
+	}
+	routeclient.Routes(jobspec.Namespace).Create(rpmroute)
+
+	watcher, err := buildclient.Builds("jobspecNamespace").Watch(meta.ListOptions{})
+	if err != nil {
+		t.Errorf("Failed to create a watcher over Builds in namespace 'jobspecNamespace'")
+	}
+	defer watcher.Stop()
+
+	clusterBehavior := func() {
```

Comment:
If this can be replaced with the following code, we don't need for the `fakeSuccessfulBuild()`.
```go
	clusterBehavior := func() {
		fakecs.buildcs.PrependWatchReactor("builds", func(action clientgotesting.Action) (bool, watch.Interface, error) {
			watcher.Stop()
			ret := watch.NewFakeWithChanSize(1, false)
			for {
				event, ok := <-watcher.ResultChan()
				if !ok {
					t.Error("Fake cluster: watcher event closed, exiting")
					break
				}
				if build, ok := event.Object.(*buildv1.Build); ok {
					t.Logf("Fake cluster: Received event on Build '%s': %s", build.ObjectMeta.Name, event.Type)
					t.Logf("Fake cluster: Updating build '%s' status to '%s' and exiting", build.ObjectMeta.Name, buildv1.BuildPhaseComplete)

					if build.Spec.Output.To.Kind == "ImageStreamTag" {
						name := build.Spec.Output.To.Name
						namespace := build.Spec.Output.To.Namespace
						t.Logf("Fake cluster: Build output is ImageStreamTag '%s' in namespace '%s': creating a fake ImageStreamTag with a fake Image", name, namespace)
						istag := &apiimagev1.ImageStreamTag{
							ObjectMeta: meta.ObjectMeta{
								Name:      name,
								Namespace: namespace,
							},
							Image: apiimagev1.Image{ObjectMeta: meta.ObjectMeta{Name: "DoesThisMatter"}},
						}
						if _, err := istclient.ImageStreamTags(namespace).Create(istag); err != nil {
							t.Errorf("Fake cluster: Failed to treate ImageStreamTag to fake Build output: %v", err)
						}
					}
					b := build.DeepCopy()
					b.Status.Phase = buildv1.BuildPhaseComplete
					ret.Modify(b)
					break
				}
				t.Logf("Fake cluster: Received non-build event: %v", event)
			}
			return true, ret, nil
		})
	}
```

either way, you understand my point.